### PR TITLE
Make more classes implement JSONSerializer to reduce boilerplate code

### PR DESCRIPTION
**Pull request by ckelleyRH**
In following packages:

com.netscape.certsrv.logging
com.netscape.certsrv.request
com.netscape.certsrv.property
com.netscape.certsrv.profile

### CODE REVIEW

File: base/common/src/test/java/com/netscape/certsrv/request/CMSRequestInfoTest.java [hunk 1]
```diff
@@ -33,7 +35,7 @@ public void testJSON() throws Exception {
         String json = before.toJSON();
         System.out.println("JSON (before): " + json);
 
-        CMSRequestInfo afterJSON = CMSRequestInfo.fromJSON(json);
+        CMSRequestInfo afterJSON = JSONSerializer.fromJSON(json, CMSRequestInfo.class);
         System.out.println("JSON (after): " + afterJSON.toJSON());
```
Comment (edewata) [inline base/common/src/test/java/com/netscape/certsrv/request/CMSRequestInfoTest.java:39]: The original JSON mapping for `CMSRequestInfo` used `JaxbAnnotationIntrospector` but we're not supposed to use it anymore. Could you verify that the old and new JSON mappings produce the same output?

Comment (ckelleyRH) [reply base/common/src/test/java/com/netscape/certsrv/request/CMSRequestInfoTest.java:39]: Seems fine, and I did similar tests for more complex mappings in #3598 
```
##JacksonAnnotationIntrospector
JSON (before): {
  "requestType" : "securityDataEnrollment",
  "requestStatus" : "complete"
}
JSON (after): {
  "requestType" : "securityDataEnrollment",
  "requestStatus" : "complete"
}

## JaxbAnnotationIntrospector
JSON (before): {
  "requestType" : "securityDataEnrollment",
  "requestStatus" : "complete"
}
JSON (after): {
  "requestType" : "securityDataEnrollment",
  "requestStatus" : "complete"
}
```

Comment (ckelleyRH) [reply base/common/src/test/java/com/netscape/certsrv/request/CMSRequestInfoTest.java:39]: I've rebased this onto #3598 so it now uses `JacksonAnnotationIntrospector`.

**Review by edewata — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**